src/nsls2api/api/v1/stats_api.py

```diff
@@ -10,21 +16,34 @@
 
 router = fastapi.APIRouter()
 
+
 @router.get("/stats", response_model=StatsModel)
 async def stats():
-    proposals = await proposal_service.proposal_count()
+    total_proposals = await proposal_service.proposal_count()
     beamlines = await beamline_service.beamline_count()
     facilities = await facility_service.facilities_count()
     commissioning = len(await proposal_service.commissioning_proposals())
 
-    faciltiy_data_health = await facility_service.is_healthy("nsls2")
+    facility_data_health = await facility_service.is_healthy("nsls2")
+
+    nsls2_proposals_per_cycle: list[ProposalsPerCycleModel] = []
+
+    nsls2_cycle_list = await facility_service.facility_cycles("nsls2")
+    for cycle in nsls2_cycle_list:
+        proposal_list = await proposal_service.fetch_proposals_for_cycle(cycle)
+        if proposal_list is not None:
+            model = ProposalsPerCycleModel(
+                cycle=cycle, proposal_count=len(proposal_list)
+            )
+            nsls2_proposals_per_cycle.append(model)
```

Comment:
This looks solid.  If this endpoint gets called a lot, it might be worth converting this for loop into a list comprehension for performance.  That can wait until if/when it is ever needed.

Reply:
I agree - was letting future-me worry about it as its fast enough for now.